ext/CMakeLists.txt

```diff
@@ -2,7 +2,12 @@ project(genesis-ext)
 
 find_package(Threads)
 
-file(ARCHIVE_EXTRACT INPUT "${CMAKE_CURRENT_SOURCE_DIR}/src.zip" DESTINATION "${CMAKE_CURRENT_SOURCE_DIR}")
+if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/src")
+    message(STATUS "SRC directory already extracted! Skipping...")
+else ()
+    message(STATUS "Extracting SRC directory!")
+    file(ARCHIVE_EXTRACT INPUT "${CMAKE_CURRENT_SOURCE_DIR}/src.zip" DESTINATION "${CMAKE_CURRENT_SOURCE_DIR}")
+endif ()
```

Comment:
I don't think you want to do this. If `ext` changes upstream (eg updating Dear ImGui, which I'm 99% sure will be required for us to use dynamic rendering), but is already extracted locally, it won't be overwritten with the new changes.

Reply:
This was more so to address an issue with extraction failing and causing cmake to have wonky behaviors due to it. Of course I can just change it to just always delete and extract as an alternative to this behavior if that would be better?

Reply:
Hmm, we should look at why that's happening. It should not be necessary to delete stuff before extracting (CMake should overwrite existing files), but if that's the hammer-only way out, so be it lol.

Reply:
The issue would only arise if you deleted your cache, but did not delete the extracted folder. I am not 100% sure why this was happening just that the above code caused the issue to stop appearing.

Reply:
I'll say for now I will update the code to just delete the extracted items then extract, but I'll also open an issue about the problem too.